openvpn/service_discovery/dto/service_proposal.go

```diff
@@ -0,0 +1,78 @@
+package dto
+
+import (
+	"github.com/mysterium/node/service_discovery/dto"
+	"encoding/json"
+	"log"
+)
+
+type proposal struct {
+	Id                int              `json:"id"`
+	Format            string           `json:"format"`
+	ServiceType       string           `json:"service_type"`
+	ProviderId        string           `json:"provider_id"`
+	PaymentMethodType string           `json:"payment_method_type"`
+	ServiceDefinition *json.RawMessage `json:"service_definition"`
+	PaymentMethod     *json.RawMessage `json:"payment_method"`
+	ProviderContacts  []dto.Contact    `json:"provider_contacts"`
+}
+
+func BuildServiceProposalFromJson(jsonData []byte) dto.ServiceProposal {
+	proposal := proposal{}
+
+	if err := json.Unmarshal([]byte(jsonData), &proposal); err != nil {
+		log.Fatal(err)
```

Comment:
Error should pushed to upper layer also `BuildServiceProposalFromJson() (dto.ServiceProposal, error)`